src/CowSdk.ts

```diff
@@ -1,13 +1,58 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
 import { version as SDK_VERSION } from '../package.json'
+import { CowApi } from './api'
+import { DEFAULT_APP_DATA_HASH } from './constants'
+import { SupportedChainId as ChainId } from './constants/chains'
 import { validateAppDataDocument } from './utils/appData'
-import { signOrder, signOrderCancellation } from './utils/sign'
+import { SdkError } from './utils/common'
+import { signOrder, signOrderCancellation, UnsignedOrder } from './utils/sign'
 
-export class CowSdk {
+export interface SdkContext {
+  appDataHash?: string
+  isDevEnvironment?: boolean
+  provider?: JsonRpcProvider
+}
+
+export class CowSdk<T extends ChainId> {
   static version = SDK_VERSION
+  chainId: T
```

Comment:
Will this be required? So plan is one chainId per network? 

What happens if the provider changes network? will this throw? Will this listen to those changes and update? If thats the case, would we really need to ask for the chainId?

Reply:
Since this PR involved interacting with the API and not any on-chain interaction I included a `signer` property and removed the provider one so we could handle later the network change event. Wdyt?